src/sensor/ping.cpp

```diff
@@ -233,7 +233,7 @@ void Ping::handleMessage(const ping_message& msg)
 
     case Ping1dId::MODE_AUTO: {
         ping1d_mode_auto m(msg);
-        if(_mode_auto != m.mode_auto()) {
+        if(_mode_auto != static_cast<bool>(m.mode_auto())) {
```

Comment:
I don't get this one. is _mode_auto a different type than m.mode_auto() ?

Reply:
yes, right now we only support "auto mode" and "manual mode" in ping1d, so this a bool is used, there's future plans to add more modes and probably that's why mode_auto returns a `uint8_t`.

Reply:
A note that this field will be reserved in the near future, and this line will be removed.